crates/tools/src/deposit_ckb.rs

```diff
@@ -0,0 +1,227 @@
+        "--privkey-path",
+        privkey_path.to_str().expect("non-utf8 file path"),
+        "--to-address",
+        address.to_string().as_str(),
+        "--capacity",
+        capacity,
+        "--tx-fee",
+        fee,
+        "--skip-check-to-address",
+    ])?;
+    let tx_hash = H256::from_str(&output.trim()[2..]).map_err(|err| err.to_string())?;
+    log::info!("tx_hash: {:#x}", tx_hash);
+
+    wait_for_tx(&mut rpc_client, &tx_hash, 180u64)?;
+
+    wait_for_balance_change(
+        &mut godwoken_rpc_client,
+        &l2_lock_hash,
+        init_balance,
+        180u64,
+    )?;
+
+    Ok(())
+}
+
+fn privkey_to_eth_address(privkey: &str) -> Result<CKBBytes, String> {
+    let privkey_data = H256::from_str(&privkey.trim()[2..]).map_err(|err| err.to_string())?;
+    let privkey = secp256k1::SecretKey::from_slice(privkey_data.as_bytes())
+        .map_err(|err| format!("Invalid secp256k1 secret key format, error: {}", err))?;
+    let pubkey = secp256k1::PublicKey::from_secret_key(&SECP256K1, &privkey);
+    let pubkey_hash = {
+        let mut hasher = Keccak256::new();
+        hasher.update(&pubkey.serialize_uncompressed()[1..]);
+        let buf = hasher.finalize();
+        let mut pubkey_hash = [0u8; 20];
+        pubkey_hash.copy_from_slice(&buf[12..]);
+        pubkey_hash
+    };
+    let s = CKBBytes::from(pubkey_hash.to_vec());
+    Ok(s)
+}
+
+fn privkey_to_lock_hash(privkey: &str) -> Result<H256, String> {
+    let privkey_data = H256::from_str(&privkey.trim()[2..]).map_err(|err| err.to_string())?;
+    let privkey = secp256k1::SecretKey::from_slice(privkey_data.as_bytes())
+        .map_err(|err| format!("Invalid secp256k1 secret key format, error: {}", err))?;
+    let pubkey = secp256k1::PublicKey::from_secret_key(&SECP256K1, &privkey);
+    let address_payload = AddressPayload::from_pubkey(&pubkey);
+
+    let lock_hash: H256 = CKBScript::from(&address_payload)
+        .calc_script_hash()
+        .unpack();
+    Ok(lock_hash)
+}
+
+// Read config.toml
+fn read_config<P: AsRef<Path>>(path: P) -> Result<Config, String> {
+    let content = fs::read(&path).map_err(|err| err.to_string())?;
+    let config = toml::from_slice(&content).map_err(|err| err.to_string())?;
+    Ok(config)
+}
+
+pub fn wait_for_balance_change(
+    godwoken_rpc_client: &mut GodwokenRpcClient,
+    from_script_hash: &H256,
+    init_balance: u128,
+    timeout_secs: u64,
+) -> Result<(), String> {
+    let retry_timeout = Duration::from_secs(timeout_secs);
+    let start_time = Instant::now();
+    while start_time.elapsed() < retry_timeout {
+        std::thread::sleep(Duration::from_secs(2));
+
+        let balance = get_balance_by_script_hash(godwoken_rpc_client, from_script_hash)?;
+        log::info!(
+            "current balance: {}, waiting for {} secs.",
+            balance,
+            start_time.elapsed().as_secs()
+        );
+
+        if balance != init_balance {
+            log::info!("deposit success!");
```

Comment:
A better way is to check whether deposit cell is consumed. But since we dont have access to ckb-indexer here, i'm ok with current way.